test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
                          :else (do
                                  (Thread/sleep 500)
                                  (recur (dec times)))))))))))
+
+
+
+(deftest ^:integration revoke-compile-master-test
+  (testing "Revoking compile master certificate"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
+        (testing "should update infrastructure CRL"
+          (let [ca-cert (bootstrap/get-ca-cert-for-running-server)
+                subject "dummy-compile-master"
+                ;; In one of the other unit tests I noticed the localhost cert is
+                ;; being revoked. If that does not affect things we could do that instead of 
+                ;; creating a new one. But wondering it actually may have impact on other 
+                ;; tests since this is integration
+                cm-cert (bootstrap/get-cert-signed-by-ca-for-running-server
+                             ca-cert
+                             subject)
+                 ;; TODO - Extract serial number of abvoe cert. Is there an easy way
+                 ;; to do it here or can we make assumption on the expected serial number
```

Comment:
https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/puppetserver/certificate_authority.clj#L1263

Reply:
Actually the q was a bit leftover as I found the util/revoked? method which seems to fit the bill (nice thing it takes in the crl file and cert).